storage/postgres/storage_unsafe.sql

```diff
@@ -0,0 +1,164 @@
+  -- X.509 certificate but is application dependent
+  leaf_value            BYTEA NOT NULL,
+  -- This is extra data that the application can associate with the leaf should it wish to.
+  -- This data is not included in signing and hashing.
+  extra_data            BYTEA,
+  -- The timestamp from when this leaf data was first queued for inclusion.
+  queue_timestamp_nanos  BIGINT NOT NULL,
+  PRIMARY KEY(leaf_identity_hash)
+);
+
+-- When a leaf is sequenced a row is added to this table. If logs allow duplicates then
+-- multiple rows will exist with different sequence numbers. The signed timestamp
+-- will be communicated via the unsequenced table as this might need to be unique, depending
+-- on the log parameters and we can't insert into this table until we have the sequence number
+-- which is not available at the time we queue the entry. We need both hashes because the
+-- LeafData table is keyed by the raw data hash.
+CREATE TABLE IF NOT EXISTS sequenced_leaf_data(
+  tree_id                   BIGINT NOT NULL,
+  sequence_number           BIGINT NOT NULL,
+  -- This is a personality specific has of some subset of the leaf data.
+  -- It's only purpose is to allow Trillian to identify duplicate entries in
+  -- the context of the personality.
+  leaf_identity_hash        BYTEA NOT NULL,
+  -- This is a MerkleLeafHash as defined by the treehasher that the log uses. For example for
+  -- CT this hash will include the leaf prefix byte as well as the leaf data.
+  merkle_leaf_hash          BYTEA NOT NULL,
+  integrate_timestamp_nanos BIGINT NOT NULL,
+  PRIMARY KEY(sequence_number)
+);
+
+CREATE INDEX SequencedLeafMerkleIdx ON sequenced_leaf_data(tree_id, merkle_leaf_hash);
+
+CREATE TABLE IF NOT EXISTS unsequenced(
+  tree_id               BIGINT NOT NULL,
+  -- The bucket field is to allow the use of time based ring bucketed schemes if desired. If
+  -- unused this should be set to zero for all entries.
+  bucket                INTEGER NOT NULL,
+  -- This is a personality specific hash of some subset of the leaf data.
+  -- It's only purpose is to allow Trillian to identify duplicate entries in
+  -- the context of the personality.
+  leaf_identity_hash    BYTEA NOT NULL,
+  -- This is a MerkleLeafHash as defined by the treehasher that the log uses. For example for
+  -- CT this hash will include the leaf prefix byte as well as the leaf data.
+  merkle_leaf_hash      BYTEA NOT NULL,
+  queue_timestamp_nanos BIGINT NOT NULL,
+  -- This is a SHA256 hash of the TreeID, LeafIdentityHash and QueueTimestampNanos. It is used
+  -- for batched deletes from the table when trillian_log_server and trillian_log_signer are
+  -- built with the batched_queue tag.
+  queue_id              BYTEA DEFAULT NULL UNIQUE,
+  PRIMARY KEY (queue_timestamp_nanos, leaf_identity_hash)
+);
+
+CREATE OR REPLACE FUNCTION public.ignore_duplicates(tree_id bigint, leaf_identity_hash bytea, leaf_value bytea, extra_data bytea, queue_timestamp_nanos bigint)
```

Comment:
how about `insert_leafdata_ignore_duplicates`?

Reply:
Sounds good to me.  When I get more time, I will reflow the pg code to eliminate the need for this check.  Like I said, I was just following the MySQL workflow because it works.